sormas-backend/src/main/java/de/symeda/sormas/backend/user/UserRoleFacadeEjb.java

```diff
@@ -123,16 +131,60 @@ public UserRoleDto getByUuid(String uuid) {
 		return toDto(userRoleService.getByUuid(uuid));
 	}
 
+	@Override
+	public UserRoleReferenceDto getReferenceByUuid(String uuid) {
+		return toReferenceDto(userRoleService.getByUuid(uuid));
+	}
+
 	@Override
 	public UserRoleDto saveUserRole(@Valid UserRoleDto dto) {
 
+		validate(dto);
+
 		UserRole entity = fromDto(dto, true);
+
 		userRoleService.ensurePersisted(entity);
 		return toDto(entity);
 	}
 
+	private void validate(UserRoleDto source) {
+		if (StringUtils.isBlank(source.getCaption())) {
+			throw new ValidationRuntimeException(I18nProperties.getValidationError(Validations.specifyCaption));
+		}
+		if (Objects.isNull(source.getJurisdictionLevel())) {
+			throw new ValidationRuntimeException(I18nProperties.getValidationError(Validations.specifyJurisdictionLevel));
+		}
+		if (!userRoleService.isCaptionUnique(source.getUuid(), source.getCaption())) {
+			throw new ValidationRuntimeException(I18nProperties.getValidationError(Validations.captionNotUnique));
+		}
+
+		Set<UserRight> userRights = source.getUserRights();
+		Set<UserRight> requiredUserRights = UserRight.getRequiredUserRights(userRights);
+		if (!userRights.containsAll(requiredUserRights)) {
+			throw new ValidationRuntimeException(
+				I18nProperties.getValidationError(
+					Validations.missingRequiredUserRights,
+					requiredUserRights.stream().filter(r -> !userRights.contains(r)).map(UserRight::toString).collect(Collectors.joining(", "))));
+		}
+
+		User currentUser = userService.getCurrentUser();
+		if (currentUser.getUserRoles().stream().anyMatch(r -> DataHelper.isSame(r, source))) {
```

Comment:
```suggestion
		if (currentUser != null && currentUser.getUserRoles().stream().anyMatch(r -> DataHelper.isSame(r, source))) {
```

e.g. on server startup we don't have a user.